### PR TITLE
Add documetation explaining basic proxy config

### PR DESCRIPTION
**Pull request by punkle**

### CODE REVIEW

File: content/docs/custom-plugins/proxy/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+You can call this API from the Backstage window context in a browser via a Custom Backstage plugin:
+
+```javascript
+fetch('/api/authenticated-proxy/my-api')
+```
+
+#### Step 3 Authenticate the Request
+
+The API configured in the Authenticated proxy target will recieve a token in the authorization bearer containing an access token. This token can be sent to the roadie Authentication system to retrieve an id representing the logged in Backstage user.
```
Comment (nicolasarnold12321) [inline content/docs/custom-plugins/proxy/index.md:77]:
```suggestion
The API configured in the Authenticated proxy target will receive a token in the authorization header containing an access token. This token can be sent to the roadie Authentication system to retrieve an id representing the logged in Backstage user.
```

File: content/docs/custom-plugins/proxy/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+```javascript
+fetch('/api/proxy/my-api')
+```
+
+## Authenticated Proxy
+
+Roadie provides a custom proxy service that can be configured to proxy through an access token minted by our own authentication system that can be validated by your third party service.
+
+This gives a custom plugin a way to interact with your own service. That service can authenticate the incoming requests from Roadie.
```
[the remark this answers is not in the document]
Comment (nicolasarnold12321) [reply content/docs/custom-plugins/proxy/index.md:49]: Actually, what do you think?

**Review by nicolasarnold12321 — COMMENTED** [review]
Minor comments

File: content/docs/custom-plugins/proxy/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (punkle) [reply content/docs/custom-plugins/proxy/index.md:49]: I think it is for custom plugins. This page is in the custom plugins section. Why do you think it's not for custom plugins?

File: content/docs/custom-plugins/proxy/index.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+---
```
Comment (Xantier) [inline content/docs/custom-plugins/proxy/index.md:1]: Maybe add a mention and a link to the IP allowlist somewhere on this page to possibly tell customers that this is the way they could self-host custom backend plugin in their own infra

**Review by Xantier — DISMISSED** [review]

File: content/docs/custom-plugins/proxy/index.md (hunk 3 above)
Comment (punkle) [reply content/docs/custom-plugins/proxy/index.md:1]: Will do thanks.

**Review by Xantier — APPROVED** [review]

**Pull request merged**